wordpress-mobile/WordPress-iOS
# Issue/4471: Remove "Private" privacy setting for Jetpack sites

Fixes #4471. The "Private" privacy setting is only supported by wpcom sites, so I've removed this setting if a site is self-hosted. I've also tidied up `SiteSettingsViewController` a little by moving some code into the existing `BlogSiteVisibilityHelper`, which I've also added unit tests to.

### To Test

1. Sign into a wpcom account with a connected Jetpack blog.
2. Select the site in the Sites list, and choose Settings (under Configure).
3. Select Privacy (under General).
4. Ensure that only the Public and Hidden options are available, and that their hint text is correct.

Also ensure that:

- For a non-Jetpack self-hosted site, the Privacy option isn't available at all (I haven't touched this, but it's worth verifying).
- Unit tests pass.

Needs review: @SergioEstevao
